### DISCUSSION TITLE
Python Django Upgrade issues

### DISCUSSION THREAD

**Discussion by samy2223** [celery/celery#8217] [Q&A] [answered]
we are in the process of upgrading python from 2.7 to 3.6.8, we seem to be having dependency issues with Django-celery since its no longer used, we currently use Django-celery 3.0.2 and since we are upgrading python we are using celery 5.1.2

As per the celery documentation i believe Database tables and modules that are related to django-celery have been depreciated and all those dependencies have been updated directly on celery version 5 where django-celery is no longer needed.

Does anyone have thoughts on what happened to specific modules and tables like djcelery_taskstate and djcelery_workerstate and djcelery.snapshot,djcelery.setup_loader(), we seem to have dependency on these modules and unable to find replacements for those.

Thank you in advance for your help

**Comment by auvipy** [comment]
you can use https://github.com/celery/django-celery-beat https://github.com/celery/django-celery-results

**Reply by samy2223** [reply]
how about djcelery.setup_loader(), it was used to discover celery task loaded for django's project, after installing celery 5 and celery-beat and celert-results, we seemed to be getting "django.core.exceptions.AppRegistryNotReady: Apps aren't loaded yet", what would be the replacement for djcelery.setup_loader.

**Reply by auvipy** [reply, marked as answer]
you should try celery 4.4.x first IMHO then celery 5.x also you should check this issue for more info https://github.com/celery/django-celery-beat/issues/151
